### ISSUE TITLE
Crash "NoClassDefFoundError"

### ISSUE THREAD

**Issue by Wohlf-craft** [gigaherz/ToolBelt#70]
I swear just before I reinstalled on my pack on Twitch (I have the latest version.) The game crashes due to 
`java.lang.NoClassDefFoundError: Lgigaherz/toolbelt/belt/ItemToolBelt;`

[](https://gist.github.com/Wohlf-craft/4da2bc72c4440ad6aba2dd96c34b8ffe)

I know it's tool belt because it was working before I re-downloaded the latest version from curseforge

I am running 1.12.2, latest version of forge (14.23.5.2854)

**Comment by gigaherz** [comment]
Can you show the log file? The latest version for 1.12 has 1.5 million downloads. No one has ever complained about it doing this, so I have to assume this crash is a consequence of some other problem.

In case your jar file somehow got corrupted, this is what it should hash as:
```
CRC32: 74B8FC22
MD5: 045CB4CB4A3C9DAF59BF4658E293CEC8
SHA-1: 1BD0829B5A762DFAF823EBCDE166DDA33B559E2F
SHA-256: 590772AFA32282299248D38C230CCDDCE3A3FE2CFA9969CBAC98F98CA55AC49A
```

**Comment by Wohlf-craft** [comment]
When I am able to get on my computer later I will. But my jar is not corrupt. I was automatically downloading the jar from curse forge according to the manifest.json. I think some of my other people were having the same issue. But using (the same version) but a older jar that didn't give me issues before fixed it. But I tried to export it into the zip and that did nothing.

**Comment by Wohlf-craft** [comment]
> Can you show the log file? The latest version for 1.12 has 1.5 million downloads. No one has ever complained about it doing this, so I have to assume this crash is a consequence of some other problem.
> 
> In case your jar file somehow got corrupted, this is what it should hash as:
> 
> ```
> CRC32: 74B8FC22
> MD5: 045CB4CB4A3C9DAF59BF4658E293CEC8
> SHA-1: 1BD0829B5A762DFAF823EBCDE166DDA33B559E2F
> SHA-256: 590772AFA32282299248D38C230CCDDCE3A3FE2CFA9969CBAC98F98CA55AC49A
> ```

I don't know how to check the hash on a jar. But if I would if I knew

**Comment by gigaherz** [comment]
I like to use a windows explorer extension called "hashtab", you can get it from [here](http://implbits.com/products/hashtab/), but there's other tools.

**Comment by Wohlf-craft** [comment]
> I like to use a windows explorer extension called "hashtab", you can get it from [here](http://implbits.com/products/hashtab/), but there's other tools.

Ok. I will check it out. It was since windows xp when I last installed WE extensions wow

**Comment by Wohlf-craft** [comment]
I can't seem to replicate the issue now, maybe something was screwing the jar. But dont close it yet.

**Comment by Wohlf-craft** [comment]
Just kidding, someone just installed the pack fresh and it did it, here's the log file
https://gist.github.com/Wohlf-craft/43e15d261935f631b84608d5a5daf366

**Comment by Wohlf-craft** [comment]
I can't compare the hashes because it's not my computer and I can't tell them to install something like that. But I may be able to try on my own.

**Comment by gigaherz** [comment]
Aha! That one does show the issue:
> java.io.FileNotFoundException: C:\Users\mathi\Twitch\Minecraft\Instances\Wohlf_pack Self Questing Edition (1)\mods\memory_repo\gigaherz\commons\gigaherz.commons-1.12.1\0.6.4\gigaherz.commons-1.12.1-0.6.4.jar (The process cannot access the file because it is being used by another process)

Something is causing the contained library "gigaherz.commons" to fail to extract, leading to
> Caused by: java.lang.ClassNotFoundException: gigaherz.common.ItemRegistered

Which is why it fails to load `ItemToolBelt`. This is either a problem in forge (possibly a race condition), or some antivirus. In either case, it should fix itself trying to load again (may take more than one try).

The only thing I can do to help is release an update that uses shading instead of contained dependencies, which could **very** slightly increase the memory usage (a few kb) if you have more than one of my 1.12 mods.

**Comment by Wohlf-craft** [comment]
uhhh
I don't have any mods other than tool belt. It's not a anti-virus, I have tested on 2 different windows machines now, mine which I use vanilla windows defender and free malwarebytes (no live malware defending) and I am not sure what the other has. It didn't do this to one of my other friends, he updated the pack and it didn't do it for him. But it did for me. It only happens when it auto-downloads the jar from curseforge. So currently, I am giving my friends the other jar that doesn't cause that issue for them to replace it. I think it has to be forge, because forge just updated to a newer version for 1.12.2, and I don't think anyone else has seen this issue or something, if they have they don't know github exists, or no one has seen the issue because they don't know forge needs updating. So I would blame it on forge. I don't know if it would fix it's self trying to reload it many times. But I will test it myself to see if it does or not.

**Comment by Wohlf-craft** [comment]
>which could **very** slightly increase the memory usage (a few kb) if you have more than one of my 1.12 mods.
as long as it's not taking more than 100 kb I am fine. Sadly, my computer can barely handle my own pack :(.

**Comment by gigaherz** [comment]
Welp maybe something broke in forge or whatever. I'll do the workaround. It's annoying but 🤷

**Comment by Wohlf-craft** [comment]
> 
> 
> Welp maybe something broke in forge or whatever. I'll do the workaround. It's annoying but 🤷

Rip

**Comment by gigaherz** [comment]
Fixed as of 1.12.2-1.9.13, should be up [on curseforge ](https://www.curseforge.com/minecraft/mc-mods/tool-belt/files/2988767)in a moment.
